src/api/v1/db/oracledb/terms-dao.js

```diff
@@ -33,10 +49,37 @@ const getPostCurrentPreInterimTermCodes = async (connection) => {
 const getTerms = async (query) => {
   const connection = await getConnection();
   try {
-    const { rows } = await connection.execute(contrib.getTerms());
     const postCurrPreTermCodes = await getPostCurrentPreInterimTermCodes(connection);
-    const serializedTerms = serializeTerms(rows, postCurrPreTermCodes, query);
-    return serializedTerms;
+
+    const { postInterimTermDate, preInterimTermDate } = query;
+
+    const postInterimTermOfDate = await getInterimTermOfDate(
+      connection, postInterimTermDate, contrib.getPostInterimTermOfDate,
+    );
+    const preInterimTermOfDate = await getInterimTermOfDate(
+      connection, preInterimTermDate, contrib.getPreInterimTermOfDate,
+    );
+
+    /**
+     * If postInterimTermOfDate and preInterimTermOfDate point to different terms, return an empty
+     * array since it's not possible to have a term that qualifies both conditions at the same time.
+     */
+    if (
+      (postInterimTermOfDate && preInterimTermOfDate)
+      && postInterimTermOfDate !== preInterimTermOfDate
+    ) {
```

Comment:
Doesn't this "postInterimTermOfDate !== preInterimTermOfDate" execute the query for both postInterimTermOfDate and preInterimTermOfDate? Which could be executed for undefined postInterimTermDate, and preInterimTermDate.